src/main/java/org/apache/solr/benchmarks/BenchmarksMain.java

```diff
@@ -309,30 +394,23 @@ static void indexJsonComplex(String baseUrl, String collection, int threads, Ind
 	 *     <li>Error count on the query (long)</li>
 	 * </ol>
 	 */
-	private static class DetailedQueryStatsListener implements ControlledExecutor.ExecutionListener<String, NamedList<Object>> {
+	private static class DetailedQueryStatsListener implements ControlledExecutor.ExecutionListener<String, SolrBenchQueryResponse> {
 		private final ConcurrentMap<String, SynchronizedDescriptiveStatistics> durationStatsByType = new ConcurrentHashMap<>();
 		private final ConcurrentMap<String, SynchronizedDescriptiveStatistics> docHitCountStatsByType = new ConcurrentHashMap<>();
 		private final ConcurrentMap<String, AtomicLong> errorCountStatsByType= new ConcurrentHashMap<>();
 		private final AtomicBoolean loggedQueryRspError = new AtomicBoolean(false);
 		private static final Logger logger = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
 		@Override
-		public void onExecutionComplete(String typeKey, NamedList<Object> queryRsp, long duration) {
+		public void onExecutionComplete(String typeKey, SolrBenchQueryResponse sbq, long duration) {
 			if (typeKey != null) {
 				SynchronizedDescriptiveStatistics durationStats = durationStatsByType.computeIfAbsent(typeKey, (key) -> new SynchronizedDescriptiveStatistics());
 				durationStats.addValue(duration / 1_000_000.0);
-				if (queryRsp != null) {
-
-					InputStream responseStream = (InputStream) queryRsp.get("stream");
-					String responseStreamAsString = "";
-					try {
-						responseStreamAsString = getResponseStreamAsString(responseStream); //should only call this once, as this reads the stream!
-					} catch (IOException e) {
-						logger.warn("Failed to read the response stream for " + typeKey);
-					}
+				if (sbq.responseString != null) {
```

Comment:
we can probably remove this check or only check if sbq is null. As `sbq.reponseString` could be null (if the response is erroneous) and in such case, we do want to increment the `errorCount`